File: kotlin/src/HelloBranch.kt
fun main() {
    var helloWorld = "Hello, world!"
    val regex = Regex("Hello")

    var timeSlot =
        if (helloWorld == "Good morning") {
            "morning"
        } else if (regex.containsMatchIn(helloWorld)) {
            "noon"
        } else {
            "night"
        }
    println(timeSlot)

    timeSlot =
        when {
            helloWorld == "Hello, world!" -> "nonn"
            helloWorld == "Good morning, world!" -> "morning"
            else -> "night"
        }
    println(timeSlot)
}

File: r/src/hello_branch.r
hello_world <- "Hello, world!"

if (hello_world == "Good morning") {
  print("morning")
} else if (grepl("Hello", hello_world, fixed = TRUE)) {
  print("noon")
} else {
  print("night")
}

time_slot <-
switch(hello_world,
  "Hello, world!" = "nonn",
  "Good morning, world!" = "morning",
  "night"
)

print(time_slot)
